crates/air-lib/trace-handler/src/data_keeper/trace_slider.rs

```diff
@@ -102,7 +105,7 @@ impl TraceSlider {
         self.position
     }
 
-    pub(crate) fn subtrace_len(&self) -> usize {
+    pub(crate) fn subtrace_len(&self) -> TraceLen {
         self.subtrace_len - self.seen_elements
```

Comment:
It worth adding a `debug_assert!(self.subtrace_len >= self.seen_elements);`.

Reply:
Not sure if this would help to catch bugs but added.

Reply:
It will document the intention.